### DISCUSSION TITLE
How to lock value field of slider?

### DISCUSSION THREAD

**Discussion by bijlevel** [RomanKubiak/ctrlr#581] [Q&A]
I want to prevent a user to enter a value in the field because the slider content is not a regular number range. How do I accomplish this? (I hope I don't overlook a setting. I couldn't find it...) TIA!

**Comment by Ted-juh** [comment]
While there is no box to be ticked to disable/ enable the textbox of a slider in "panel mode" of Ctrlr, it should be accessible through Lua. See: LComponents.cpp under LSlider::wrapForLua it gives the def:
.def("setTextBoxIsEditable", &Slider::setTextBoxIsEditable)
Which should be a boolean. 

I need some time to figure out how to implement it. It's been a while since I poked around in the Slider Class and got things working that weren't available through the "panel mode" of Ctrlr. 

Or, block the textbox with another component on top? Or display the value in a label, and set the "noTextBox" option for the Slider. Just a few thoughts.

**Comment by dobo365** [comment]
Best and simplest is indeed the last option mentioned by Tedjuh: display the value in a separate label, and set the "noTextBox" option for the Slider.

**Comment by bijlevel** [comment]
Thanks for these valuable tips! Indeed the "display value in label/"noTextBox" option" seems promising, but do I have to set the "noTextBox" via Lua programming?

**Comment by Ted-juh** [comment]
Nope, it's an option when in panel mode for the slider. Somewhere at three quarter of the options. It's now probably set to "textbox below".

**Reply by bijlevel** [reply]
Ah I see! I never paid attention to that specific slider option, because until now I never needed it ;-). Thanks/Dank u!
